src/cpu.h

```diff
@@ -1,55 +1,64 @@
-#ifndef CPU_H
-#define CPU_H
-#include <stdint.h>
+#ifndef YOBEMAG_CPU_H
+#define YOBEMAG_CPU_H
 
-typedef struct cpu {
-    uint8_t H;
-    uint8_t L;
+#include <stdint.h>
 
-    uint8_t D;
-    uint8_t E;
+typedef enum flag {
+	/*
+	 * Carry flag
+	 */
+	C_FLAG = 4,
+	/*
+	 * Half carry flag
+	 */
+	H_FLAG = 5,
+	/*
+	 * Subtract flag
+	 */
+	N_FLAG = 6,
+	/*
+	 * Zero flag
+	 */
+	Z_FLAG = 7,
+} Flag;
 
-    uint8_t B;
-    uint8_t C;
+typedef union Register {
+	struct {
+		uint8_t high; // A, D, D, H
+		uint8_t low; // F, C, E, L
+	} bytes;
+	uint16_t word;
+} Register;
```

Comment:
Is using upper case common standard for setting "types" in C?

Reply:
Google, LLVM, etc. do this, i.e., PascalCase
GNU is the only one doing snake_case

Reply:
I see, thx